commands/bnet.js

```diff
@@ -0,0 +1,443 @@
+const Discord = require('discord.js');
+
+const kCharsPerMessage = 1000;
+const kHeaderString = 'This channel stores a community collection of ' +
+'BattleTags for quick reference.\nUse ' +
+'\'[prefix] bnet [add|remove] [battleTag]\' in any channel to add or remove ' +
+'a BattleTag from your account.\n';
+
+const kSnowflakeRegex = new RegExp(/<@(\d+)>/);
+const kBattleTagRegex = new RegExp(/.{3,12}#\d{4,5}/);
+const kBattleTagOnlyRegex = new RegExp('^' + kBattleTagRegex.source + '$');
+const kBattletagLineRegex = new RegExp('(.*): (' + kBattleTagRegex.source +
+                                       ', )*('+ kBattleTagRegex.source+ ')');
+
+let channelId = null;
+let battleTags = null;
+let modRoles = null;
+
+/**
+ * A class to store a set of BattleTags for a particular user or text key. Only
+ * one of user and textId will be set. If the class is constructed with a
```

Comment:
Why not just use the user everywhere? I think it'd make some things simpler. The only use case I can think of using textId is for adding people not in the discord which I feel like shouldn't be a thing anyways.

Reply:
Fair enough. Done.